### DISCUSSION TITLE
Backup  of data (in the context of mzXML files)

### DISCUSSION THREAD

**Discussion by hepcat72** [Princeton-LSI-ResearchComputing/tracebase#1370] [Ideas]
## Currently, data exists in these locations:
- `/tracebase-submissions` (the ms data share)
  - backed up
  - user editable
  - no change history
  - mzXML
    - all
  - raw
    - all
- `/tracebase-archive` (prod and dev are separate location) **backed up, but not everything^ and no change history**
  - backed up
  - NOT user editable
  - no change history
  - mzXML
    - excludes
      - blanks
      - unanalyzed
      - files with no scans
  - No raw files
- `/tracebase-staging` (temporary)
- `tracebase-rabinowitz-data` (repo) **essentially a backup and has a change history, but no mzXML**
  - backed up
  - NOT user editable
  - change history
  - No mzXML
  - No raw files

## Current Issues

`/tracebase-archive` is our current backup plan for maintaining mzXML files, but it does not include everything.  It is missing blanks, unanalyzed files, and files with no scan tags.

Empty `mzXML`s (i.e. those with no scan tags) are never anticipated to be an issue, but the others are associated with some reasonable concerns:

- Someone in the future may reasonably want to compare an existing sample with a blank, but that data may get lost.
- Users may want to revisit previously unanalyzed `mzXML` files, but that data could currently get lost.
  - `mzXML` files that were not analyzed (included in a peak annotation file) fall into 2 categories
    - **Those that will eventually be analyzed** (i.e. were just overlooked).  There are numerous examples of these in the current data.
    - **Those that were skipped with no intention to analyze them.**

As far as I see it, one of the primary reasons for TraceBase was to be able to find unanalyzed data, which is the main reason I thought we should be loading these files.  A concern was raised about the value of loading data that is not linked to any metadata in tracebase, so let me clarify from the perspective of the usage of the term "orphaned" that was used during the meeting...

An mzXML file, as an `ArchiveFile` record, can be linked via an `MSRunSample` record *if* we have a `Sample` record and an `MSRunSequence` record.  Users will have this info.  It's only missing from the Samples (and possibly Animals) sheet because the samples were stubbed out using the peak annotation files.  All I'm saying is that we should have them fill it out for these unanalyzed files and load them.  Then they'll be backed up.

If the user overlooked, for example, the negative mzXML files, but did the positive ones, the samples and sequences will already exist.

The file names have the sample names in them, and those samples are associated with animals and studies, which have all the metadata needed for the mzXML to be useful.  If there are any `mzXML` files for which they cannot remember the animals or even the study, despite the file names and the study directory we found them in, then I agree, we don't need to back those up.  We need the metadata for the `mzXML`s to be of any use.  I doubt that we will encounter this.

**Comment by mneinast** [comment]
thanks for outlining this - it's helped me understand where these files are right now.

Would it help to have a new directory, `submission-snapshots`, which copies every file in a submission directory when the user has finally completed the form / email to submit?  And lives forever?  Maybe this would still be manually copied by developers when they get around to looking at the study, but it seems like a useful backup / in-between solution.
